### ISSUE TITLE
Start of audio missing if SoundTouch is enabled for AudioFilePlayerExt

### ISSUE THREAD

**Issue by jamiebullock** [drowaudio/drowaudio#7]
Steps to reproduce:

- Create an instance of `AudioFilePlayerExt`
- Add relevant `getNextAudioBlock()` and `prepareToPlay()` calls to audio processor
- call `setInputStream()` with some valid stream (the bug manifests regardless of stream type)
- call `start()` e.g. from a "play' button
- Compile and run

Expected:

- When play button is pressed playback starts from beginning of audio stream

Actual:

- Playback starts, but the beginning of the stream is missing

Removing the `SoundTouchAudioSource` from the audio graph resolves the problem, so the issue is being introduce by `SoundTouchAudioSource`. It's like it's causing a 100ms or so "fade in" on the stream. 

Any ideas how I might fix this?

**Comment by jamiebullock** [comment]
OK, after a bit more digging on this, it _seems_ like `SoundTouchAudioSource` produces silence for the first `numberOfSamplesToBuffer` samples, so for a stereo source this overwrites the first `0.5 * numberOfSamplesToBuffer / sampleRate` seconds of a stream with perceptual silence (although it doesn't appear to be actually writing zeroes.

As a workaround, this can therefore be compensated for by calling `setPosition()` with something like `desiredPositionInSeconds - (0.5 * numberOfSamplesToBuffer / sampleRate)` prior to calling start()

I couldn't figure out why was causing the problem in the first place though. If it was simply a delay for the `SoundTouchAudioSource` to buffer, I would simply expect the stream to be _delayed_ not to have the start of the stream overwritten.

**Comment by jamiebullock** [comment]
This turns out to have been an upstream bug in SoundTouch. Fixed in r244 https://sourceforge.net/p/soundtouch/code/244/

**Comment by jamiebullock** [comment]
Actually, it seems I closed this prematurely. The above mentioned fix doesn't actually resolve the issue, although I guess this may still be an upstream bug.

**Comment by drowaudio** [comment]
Yes, I think the bug you listed is unrelated. It is simply an algorithm drawback of SoundTouch that the first ~100ms of audio is silent due to needing a certain number of samples to process.

However, I'm not sure if this 100ms is a hard value. It seems to be dependant on the type of processing used (e.g. if time-stretching is employed which is used n pitch or tempo adjustments but not speed). It may also vary with the use of other parameters e.g. the quickseek parameter.

If you can find some anecdotal evidence perhaps this can be baked in to the SoundTouchAudioProcessor when the stream is reset?
